pkg/member/member_control.go

```diff
@@ -42,6 +44,9 @@ type Control interface {
 	// AddMemberAsLearner add a new member as a learner to the etcd cluster
 	AddMemberAsLearner(context.Context) error
 
+	// AddLearnerWithRetry add a new member as a learner with exponential backoff.
+	AddLearnerWithRetry(context.Context, int, string) error
```

Comment:
Control interface is a low level interface and should not have this method. It is the responsibility of the caller to wrap it with retry logic. Create a util function - you can also take inspiration from https://github.com/gardener/dependency-watchdog/blob/master/internal/util/retry.go

Reply:
why to create a retry func when we can reuse this `"k8s.io/client-go/util/retry"` ?

Reply:
I can move `AddLearnerWithRetry ` out of this interface to different pkg

Reply:
>why to create a retry func when we can reuse this "k8s.io/client-go/util/retry"

Sure you can use util.retry as you are using today. I only gave that as an example to have a utility function to do retries.